File: android/src/main/java/com/saltechsystems/couchbase_lite/DatabaseEventListener.java
package com.saltechsystems.couchbase_lite;

import io.flutter.plugin.common.EventChannel;

public class DatabaseEventListener implements EventChannel.StreamHandler {
    public EventChannel.EventSink mEventSink;

    /*
     * IMPLEMENTATION OF EVENTCHANNEL.STREAMHANDLER
     */

    @Override
    public void onListen(Object args, final EventChannel.EventSink eventSink) {
        mEventSink = eventSink;
    }

    @Override
    public void onCancel(Object args) {
        mEventSink = null;
    }
}

File: ios/Classes/DatabaseEventListener.swift
//
//  DatabaseEventListener.swift
//  couchbase_lite
//
//  Created by Kin Mak on 8/7/2020.
//

import Foundation
import CouchbaseLiteSwift

class DatabaseEventListener: FlutterStreamHandler {
    var mEventSink: FlutterEventSink?
    
    func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        mEventSink = events
        
        return nil
    }
    
    func onCancel(withArguments arguments: Any?) -> FlutterError? {
        
        
        return nil
    }
}
